addon/components/polaris-resource-list.js

```diff
@@ -28,809 +28,677 @@ const LARGE_SPINNER_HEIGHT = 45;
-
-        if (!this.get('selectable') || !hasMoreItems) {
-          return;
-        }
-
-        let actionText =
-          selectedItems === SELECT_ALL_ITEMS
-            ? 'Undo'
-            : `Select all ${items.length}+ ${get(
-                resourceName,
-                'plural'
-              )} in your store`;
-
-        return {
-          content: actionText,
-          onAction: () => this.handleSelectAllItemsInStore,
-        };
-      }
-    ).readOnly(),
-
-    emptySearchResultTitle: computed('resourceName.plural', function() {
-      let resourceName =
-        this.get('resourceName') || this.get('defaultResourceName');
-
-      return `No ${get(resourceName, 'plural')} found`;
-    }).readOnly(),
-
-    showEmptyState: computed(
-      'filterControl',
-      'itemsExist',
-      'loading',
-      function() {
-        let { filterControl, itemsExist, loading } = this.getProperties(
-          'filterControl',
-          'itemsExist',
-          'loading'
-        );
-
-        return filterControl && !itemsExist && !loading;
-      }
-    ).readOnly(),
-
-    spinnerStyle: computed('loadingPosition', function() {
-      let loadingPosition = this.get('loadingPosition');
-      let defaultTopPadding = 8;
-      let topPadding =
-        loadingPosition > 0 ? loadingPosition : defaultTopPadding;
-      return htmlSafe(`padding-top: ${topPadding}px`);
-    }).readOnly(),
-
-    spinnerSize: computed('items.length', function() {
-      return this.get('items.length') < 2 ? 'small' : 'large';
-    }).readOnly(),
-
-    context: computed(
-      'selectedItems.[]',
-      'resourceName',
-      'loading',
-      'selectMode',
-      'selectable',
-      function() {
-        let {
-          selectedItems,
-          resourceName,
-          loading,
-          selectMode,
-          selectable,
-        } = this.getProperties(
-          'selectedItems',
-          'resourceName',
-          'loading',
-          'selectMode',
-          'selectable'
-        );
-        resourceName = resourceName || this.get('defaultResourceName');
-        return {
-          selectable,
-          selectedItems,
-          selectMode,
-          resourceName,
-          loading,
-          onSelectionChange: (...args) => {
-            return this.handleSelectionChange(...args);
-          },
-        };
-      }
-    ).readOnly(),
-
-    handleResize() {
-      // This is needed to replicate the React implementation's `@debounce` decorator.
-      this.debounceTask('debouncedHandleResize', 0);
-    },
-
-    debouncedHandleResize() {
-      // In the React implementation, the resize event listener
-      // is only rendered when `selectable` is truthy. We handle
-      // that by bombing out of this handler when that condition
-      // is met rather than dynamically managing event listeners.
-      if (!this.get('selectable')) {
-        return;
-      }
+  @(computed('selectable', 'sortOptions.length', 'alternateTool').readOnly())
```

Comment:
Do we need the `.readOnly()` here (and in some other places) if this CP is defined without a setter? 🤔

Reply:
Hm this was all codemod magic, we've been keeping them in other branches. @vladucu any strong feelings on dropping `readOnly` or not for CP's without setters?

Reply:
I have no strong opinion here.....I have not commented on these in any PRs, just for the sake of getting these tasks done asap

we'll do some more tweaks at some point in the near future if we want to move to glimmer components with the new tracking feature and we can do at that point too